src/pykiso/lib/connectors/cc_pcan_can.py

```diff
@@ -135,26 +138,35 @@ def __init__(
         self.bus = None
         self.logging_activated = logging_activated
         self.raw_pcan_interface = None
-        self.logging_path = None
         # Set a timeout to send the signal to the GIL to change thread.
         # In case of a multi-threading system, all tasks will be called one after the other.
         self.timeout = 1e-6
+        self.trc_count = 0
+
         """
-        Extract the base logging directory from the logging module, so we can
-        create our logging folder in the correct place.
-        logging_path will be set to the parent directory of the logfile which
-        is set in the logging module + /raw/PCAN
-        If an AttributeError occurs, the logging module is not set to log into
-        a file.
-        In this case we set the logging_path to None and will just log into a
-        generic logfile in the current working directory, which will be
-        overwritten every time, a log is initiated.
+        Logging in the trace will fall under those cases:
+        1 - trace_path is a .trc file, CAN messages will be logged in it. If it already
+            exists If it already exists, it will be overwritten.
+        2 - trace_path has another file extension, a ValueError will be raised.
+        3 - trace_path is an existing folder, a default name will be generated
+            for the trace file containing the timestamp.
+        4 - trace_path is a non-existent folder, in this case the folder will be
+            created and the default name will be used for the trace file.
+        5 - trace_path is not defined by the user, in this case the trace_path is set to
+        None and log will be saved in the default file in the current working directory.
         """
-        if self.trace_path.is_file():
-            self.trace_path = self.trace_path.parent
-            log.internal_warning(
-                f"File names are not supported for trace file creation. Trace will be written to {self.trace_path}"
+        if self.trace_path == "":
```

Comment:
I'm a bit confused with the flow here. Could you please have:
1. Best case
2. ...
3. ...
4. Error handling
?

Reply:
In addition, codecov is complaining of the covereage

Reply:
In my opinion this text block has no reason to be here (it isn't even considered as a real docstring)

It would be better to simply extend the explanation of the difference trace-specific parameters in the constructor

Reply:
Done